core/reactor_common.c

```diff
@@ -1292,17 +1300,32 @@ trigger_handle_t _lf_schedule_int(lf_action_base_t* action, interval_t extra_del
 }
 
 /**
+
  * Invoke the given reaction
  *
  * @param reaction The reaction that has just executed.
  * @param worker The thread number of the worker thread or 0 for unthreaded execution (for tracing).
  */
 void _lf_invoke_reaction(reaction_t* reaction, int worker) {
+
+#ifdef LF_THREADED
+    if (((self_base_t*) reaction->self)->reactor_mutex != NULL) {
```

Comment:
We've seen a increase this kind the use of preprocessor directives inside of function bodies throughout the code base, and this has me worried about code maintainability. Here is an excerpt from the Linux kernel coding style:

> Wherever possible, don’t use preprocessor conditionals (#if, #ifdef) in .c files; doing so makes code harder to read and logic harder to follow. Instead, use such conditionals in a header file defining functions for use in those .c files, providing no-op stub versions in the #else case, and then call those functions unconditionally from .c files. The compiler will avoid generating any code for the stub calls, producing identical results, but the logic will remain easy to follow.

In short, I think we should stop using this anti pattern. A good start with this would be try to stop adding more occurrences of it, at least inside of function bodies.

Reply:
Two possible solutions:
1. Implement `void lf_invoke_reaction` in both `reactor.c` and `reactor_threaded.c` which eventually calls `_lf_invoke_reaction` in `reactor_common.c`. Then we can lock the mutex in the version implement in `reactor_threaded.c`
2. Just put the mutex pointer on the self struct, also for the unthreaded runtime. This, of course, adds additional memory footprint for the unthreaded runtime which we do intend for resource constraint platforms...

Reply:
We can address this in a separate PR as @edwardalee suggested. I've created an issue https://github.com/lf-lang/reactor-c/issues/210 for it.